src/OmniSharp.Roslyn.CSharp/Services/Intellisense/CompletionItemExtensions.cs

```diff
@@ -60,9 +60,13 @@ public static async Task<IEnumerable<ISymbol>> GetCompletionSymbolsAsync(this Co
             }
 
             // if the completion provider encoded symbols into Properties, we can return them
-            if (properties.ContainsKey(SymbolName) && properties.ContainsKey(SymbolKind))
+            if (properties.TryGetValue(SymbolName, out string symbolNameValue)
+                && properties.TryGetValue(SymbolKind, out string symbolKindValue)
```

Comment:
Calls to the ImmutableDictionary.GetValue<> were the main CPU-hog, now they are cached outside of the loop.